Pipeline: output: opensearch: style (#1961)
fluent/fluent-bit-docs: pipeline/outputs/opensearch.md

# OpenSearch

> The _OpenSearch output plugin lets you ingest your records into an [OpenSearch](https://opensearch.org/) database.

Reviewer: ```suggestion
The _OpenSearch_ output plugin lets you ingest your records into an [OpenSearch](https://opensearch.org/) database.
```

> The following instructions assumes that you have an operational OpenSearch service running in your environment.

Reviewer: ```suggestion
The following instructions assume you have an operational OpenSearch service running in your environment.
```

For records that do nor have the field `kubernetes.namespace_name`, the default prefix, `logstash` will be used.

### Fluent Bit and Amazon OpenSearch Service

The Amazon OpenSearch Service adds an extra security layer where HTTP requests must be signed with AWS Sigv4. This plugin supports Amazon OpenSearch Service with IAM Authentication.

> See [AWS credentials](../../administration/aws-credentials.md) for details on how AWS credentials are fetched.

Reviewer: ```suggestion
For details about how AWS credentials are fetched, see [AWS credentials](../../administration/aws-credentials.md).
```

...
```

This behavior could be indicative of a hard-to-detect issue with index shard usage in the OpenSearch domain. While OpenSearch index shards and disk space are related, they're not directly tied to one another. OpenSearch domains are limited to 1000 index shards per data node, regardless of the size of the nodes. Shard usage isn't proportional to disk usage: an individual index shard can hold anywhere from a few kilobytes to dozens of gigabytes of data.

> Depending index creation and shard allocation configuration in the OpenSearch domain, all the available index shards could be used before the data nodes run out of disk space. This can result in exhibition disk-related performance issues (nodes crashing, data corruption, or the dashboard going offline). The primary issue that arises when a domain is out of available index shards is that new indexes can no longer be created (though logs can still be added to existing indexes).

Reviewer: ```suggestion
Depending index creation and shard allocation configuration in the OpenSearch domain, all the available index shards could be used before the data nodes run out of disk space. This can result in exhibition disk-related performance issues (nodes crashing, data corruption, or the dashboard going offline). The primary issue that arises when a domain is out of available index shards is that new indexes can no longer be created (although logs can still be added to existing indexes).
```

> When that happens, the Fluent Bit OpenSearch output can show confusing behavior. For
> example:

Reviewer: ```suggestion
When that happens, the Fluent Bit OpenSearch output can show confusing behavior. For example:
```

GET _cat/allocation?v
```

> Index creation issues can appear if any hot data nodes have around 1000 shards. Creation issues can also occur if the total number of shards spread across hot and ultra warm data nodes in the cluster is greater than 1000 times the total number of nodes. For example, in a cluster with 6 nodes, the maximum shard count would be `6000`.

Reviewer: ```suggestion
Index creation issues can appear if any hot data nodes have around 1,000 shards. Creation issues can also occur if the total number of shards spread across hot and ultra warm data nodes in the cluster is greater than 1,000 times the total number of nodes. For example, in a cluster with six nodes, the maximum shard count would be `6000`.
```
